src/plonk/permutation.rs

```diff
@@ -0,0 +1,14 @@
+//! Implementation of a PLONK permutation argument.
+
+use crate::arithmetic::CurveAffine;
+
+mod prover;
+mod verifier;
+
+#[derive(Debug, Clone)]
+pub(crate) struct Proof<C: CurveAffine> {
```

Comment:
This currently uses a single permutation "proof" to represent all the permutations added to the circuit. This was the most straight-forward refactoring of the existing code (making review of the moving pieces easier), but we will probably want to alter this in a subsequent PR to instead have a proof per permutation (similar to the lookup argument PR).

Reply:
I've now extracted the permutation keygen logic as well, and I've done so in the same way as the lookup argument (`Vec<permutation::Argument>`).